### DISCUSSION TITLE
dhcp module not working on ubuntu 22.04

### DISCUSSION THREAD

**Discussion by penguinpages** [webmin/webmin#1873] [General]
I am not sure if this is a webmin  cause or ubuntu.  But been trying for days now to just get a basic DHPCD service to start to host subnet

Existing server is Ubuntu 20 with webmin and working

New servers are to be a cluster pair with HA DHCP..  but even with copy over of existing dhcpd.conf ..  dhcpd will not start..  nor with a VERY basic click though to host single subnet

I have a more lenghthy posting here  https://askubuntu.com/questions/1460924/isc-dhcp-server-ubuntu-22-04-no-subnet-declaration    but hoping this has been something webmin team has seen or has in put into it.

**Comment by swelljoe** [comment]
Are you using Webmin to manage dhcpd? If not, it can't possibly have anything to do with your problem. And, I would recommend you try it without Webmin, so you can reduce the number of variables.

You'll need to look at the relevant log entries to find out _why_ it won't start (and, again, it's extremely unlikely Webmin has anything to do with why it won't start, but you can rule it out as a variable quite easily by trying it without Webmin involved).

**Comment by penguinpages** [comment]
Yes.. 100% webmin.  The goal is to create a GUI / user based friendly (maybe Terraform / ansible :) controlled means to manage and get subnets created.

GUI for now..   HA ... then layer in IaC

**Comment by swelljoe** [comment]
I'm not saying you cannot use Webmin to configure/manage it. I am saying you need to reduce the number of variables because your assessment of the problem is too complicated. You need to figure out the reason it won't start. Removing Webmin from the equation removes a variable.

**Comment by penguinpages** [comment]
ya.. I can't blame webmin.  But I was hoping someone else with basic Ubuntu 22.04 with "test subnet" noted..  ... stupid simple.. failing to start..  maybe this is something new to dhcpd service

**Comment by penguinpages** [comment]
Found fix.   But it is a bug IMAO.  I will post to 

Fix:  You must create a scope even if it has no ips within scope to lease out.   
Ex:  DNS / DHCP servers are on subnet that has static IPs for servers.  This subnet has no need for DHCP. But the server being on that subnet still must have decleration of that IP subnet
10.89.100.152 is host ip.  I had to create scope 10.89.100.0/24 with no ips to lease out.  Then dhcp would start.    Even though router helper services would never have reference to it , and it will never hand out IPs on that subnet.

Maybe webmin team could note this?  Setup scope based on NIC selected to listen to as check to save?
